reaktive/src/commonMain/kotlin/com/badoo/reaktive/maybe/ZipWith.kt

```diff
@@ -0,0 +1,7 @@
+package com.badoo.reaktive.maybe
+
+fun <T, R> Maybe<T>.zipWith(other: Maybe<R>): Maybe<Pair<T, R>> =
```

Comment:
Do you think we really need this variant in the library? You could just write: `a.zipWith(b, ::Pair)`.

Reply:
I've been thinking about it.
So I think having zipWith(other) is useful in that it encourages people to use it without a zipper function. That is, it highlights that you can just combine two requests as a pair. It also reads slightly nicer.
```suggestion
fun loadSomething() =
    dataSource.loadSome()
        .zipWith(dataSource.loadThing())

fun loadSomething() =
    dataSource.loadSome()
        .zipWith(dataSource.loadThing(), ::Pair)
```
However, this is so small I don't think it really matters either way. I'll let you decide.

Reply:
Cool, then I don't think we should have such a simple shortcuts in the library :-)